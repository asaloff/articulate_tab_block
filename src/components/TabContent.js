import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TabContentElementContainer from '../containers/TabContentElementContainer';

class TabContent extends Component {
  constructor() {
    super();
    this.state = { visible: false };

    this.makeVisible = this.makeVisible.bind(this);
    this.handleImageResize = this.handleImageResize.bind(this);
  }

  componentDidMount() {
    this.makeVisible();
  }

  componentWillReceiveProps() {
    this.setState({ visible: false });
  }

  componentDidUpdate() {
    if (!this.state.visible) {
      this.makeVisible();
    }
  }

  makeVisible() {
    setTimeout(() => {
      this.setState({ visible: true });
    }, 0);
  }

  handleImageResize() {
    const { removeFullPageImage } = this.props;
    const images = document.getElementsByTagName('img');
    for (let i = 0; i < images.length; i++) images[i].classList.remove('selected');
    removeFullPageImage();
  }

  render() {
    const { selectedTab, fullPageImage } = this.props;

    return (
      <div className={`TabContent ${ this.state.visible ? 'visible' : '' }`}>
        {selectedTab.content.map((content, index) => (
          <TabContentElementContainer key={index} content={content} />
        ))}
        <span
          className={`backdrop ${fullPageImage ? 'cover' : ''}`}
          onClick={fullPageImage && this.handleImageResize}
        >
        </span>
      </div>
    );
  }
}

TabContent.propTypes = {
  selectedTab: PropTypes.object.isRequired,
  fullPageImage: PropTypes.bool.isRequired,
  removeFullPageImage: PropTypes.func.isRequired
};

export default TabContent;
