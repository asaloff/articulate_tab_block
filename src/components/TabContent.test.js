import React from 'react';
import { shallow } from 'enzyme';
import TabContent from './TabContent';
import tabData from '../tabs';

describe('TabContent', () => {
  const props = { selectedTab: tabData.tabs[0], fullPageImage: false, removeFullPageImage: jest.mock };
  const tabContent = shallow(<TabContent {...props} />);

  it('renders properly', () => expect(tabContent).toMatchSnapshot());

  it('renders a connected TabContentElement component', () => {
    expect(tabContent.find('Connect(TabContentElement)').exists()).toBe(true);
  });
});
