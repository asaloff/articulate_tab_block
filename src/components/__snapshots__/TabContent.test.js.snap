// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContent renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabContent
    fullPageImage={false}
    removeFullPageImage={[Function]}
    selectedTab={
        Object {
            "content": Array [
              Object {
                "content": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
                "tag": "p",
              },
              Object {
                "content": "https://cdn.articulate.com/rise/courses/_Af0P0L1E-1akg7PhqRPNyg0uRFD0pUp/q0r7xIVMCo4RkD5A.gif",
                "tag": "img",
              },
            ],
            "id": 1,
            "title": "Embracing Discovery",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Connect(TabContentElement)
            content={
                        Object {
                                    "content": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
                                    "tag": "p",
                                  }
            }
/>,
          <Connect(TabContentElement)
            content={
                        Object {
                                    "content": "https://cdn.articulate.com/rise/courses/_Af0P0L1E-1akg7PhqRPNyg0uRFD0pUp/q0r7xIVMCo4RkD5A.gif",
                                    "tag": "img",
                                  }
            }
/>,
        ],
        <span
          className="backdrop "
          onClick={false}
/>,
      ],
      "className": "TabContent ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "content": Object {
            "content": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
            "tag": "p",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "content": Object {
            "content": "https://cdn.articulate.com/rise/courses/_Af0P0L1E-1akg7PhqRPNyg0uRFD0pUp/q0r7xIVMCo4RkD5A.gif",
            "tag": "img",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "backdrop ",
          "onClick": false,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Connect(TabContentElement)
              content={
                            Object {
                                          "content": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
                                          "tag": "p",
                                        }
              }
/>,
            <Connect(TabContentElement)
              content={
                            Object {
                                          "content": "https://cdn.articulate.com/rise/courses/_Af0P0L1E-1akg7PhqRPNyg0uRFD0pUp/q0r7xIVMCo4RkD5A.gif",
                                          "tag": "img",
                                        }
              }
/>,
          ],
          <span
            className="backdrop "
            onClick={false}
/>,
        ],
        "className": "TabContent ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "content": Object {
              "content": "Every creative endeavor requires that you take risks. If you try and don't succeed, you've still learned something. It took Thomas Edison more than 10,000 tries to invent a viable lightbulb. You're not failing. You're discovering what doesn't work.",
              "tag": "p",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "content": Object {
              "content": "https://cdn.articulate.com/rise/courses/_Af0P0L1E-1akg7PhqRPNyg0uRFD0pUp/q0r7xIVMCo4RkD5A.gif",
              "tag": "img",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "backdrop ",
            "onClick": false,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
